src/main/java/appeng/me/storage/CellInventory.java

```diff
@@ -442,21 +444,58 @@ protected void loadCellItems()
 
 		final int types = (int) this.getStoredItemTypes();
 
-		for( int x = 0; x < types; x++ )
+		for( int slot = 0; slot < types; slot++ )
 		{
-			final ItemStack t = ItemStack.loadItemStackFromNBT( this.tagCompound.getCompoundTag( itemSlots[x] ) );
-			if( t != null )
+			NBTTagCompound compoundTag = this.tagCompound.getCompoundTag( itemSlots[slot] );
+			int stackSize = this.tagCompound.getInteger( itemSlotCount[slot] );
+			loadCellItem( compoundTag, stackSize );
+		}
+
+		// cellItems.clean();
+	}
+
+	private void loadCellItem( NBTTagCompound compoundTag, int stackSize )
+	{
+
+		// Now load the item stack
+		final ItemStack t;
+		try
+		{
+			t = ItemStack.loadItemStackFromNBT( compoundTag );
+			if( t == null )
+			{
+				AELog.warn( "Removing item " + compoundTag + " from storage cell because the associated item type couldn't be found." );
+				return;
+			}
+		}
+		catch( Throwable ex )
+		{
+			if( AEConfig.instance.removeCrashingItemsOnLoad )
 			{
-				t.stackSize = this.tagCompound.getInteger( itemSlotCount[x] );
+				AELog.warn( ex, "Removing item " + compoundTag + " from storage cell because loading the ItemStack crashed." );
+				return;
+			}
+			throw ex;
+		}
+
+		t.stackSize = stackSize;
 
-				if( t.stackSize > 0 )
+		if( t.stackSize > 0 )
+		{
+			try
+			{
+				this.cellItems.add( AEItemStack.create( t ) );
+			}
+			catch( Throwable ex )
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Catch Exception instead of Throwable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1181)